From 6f2f6adfc442067833cef0e3391be3d77d9d122c Mon Sep 17 00:00:00 2001
From: jhenrique09 <jhsv09@gmail.com>
Date: Fri, 4 Nov 2022 15:40:04 -0300
Subject: [PATCH] PIHooks: Get package name from context instead of app

11-04 08:48:39.039 11637 11637 E AndroidRuntime: FATAL EXCEPTION: main
11-04 08:48:39.039 11637 11637 E AndroidRuntime: Process: com.NextFloor.DestinyChild, PID: 11637
11-04 08:48:39.039 11637 11637 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider androidx.startup.InitializationProvider: androidx.startup.StartupException: android.content.pm.PackageManager$NameNotFoundException: ComponentInfo{/androidx.startup.InitializationProvider}
11-04 08:48:39.039 11637 11637 E AndroidRuntime:   at android.app.ActivityThread.installProvider(ActivityThread.java:7488)

Change-Id: Icb12f938fe0fca710f8f9d29182d0134ba3c63eb
[ghostrider-reborn: adapt to PIHooks]
Signed-off-by: Adithya R <gh0strider.2k18.reborn@gmail.com>
Signed-off-by: penglezos <panagiotisegl@gmail.com>
---
 core/java/android/app/Instrumentation.java             |  4 ++--
 .../com/android/internal/util/PropImitationHooks.java  | 10 ++++++----
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/core/java/android/app/Instrumentation.java b/core/java/android/app/Instrumentation.java
index 221e34f1df8c..e17d6805700a 100644
--- a/core/java/android/app/Instrumentation.java
+++ b/core/java/android/app/Instrumentation.java
@@ -1243,7 +1243,7 @@ public class Instrumentation {
         Application app = getFactory(context.getPackageName())
                 .instantiateApplication(cl, className);
         app.attach(context);
-        PropImitationHooks.setProps(app);
+        PropImitationHooks.setProps(context);
         return app;
     }
     
@@ -1261,7 +1261,7 @@ public class Instrumentation {
             ClassNotFoundException {
         Application app = (Application)clazz.newInstance();
         app.attach(context);
-        PropImitationHooks.setProps(app);
+        PropImitationHooks.setProps(context);
         return app;
     }
 
diff --git a/core/java/com/android/internal/util/PropImitationHooks.java b/core/java/com/android/internal/util/PropImitationHooks.java
index 54ad67da524d..376ef74b4abd 100644
--- a/core/java/com/android/internal/util/PropImitationHooks.java
+++ b/core/java/com/android/internal/util/PropImitationHooks.java
@@ -17,8 +17,10 @@
 package com.android.internal.util;
 
 import android.app.Application;
+import android.content.Context;
 import android.content.res.Resources;
 import android.os.Build;
+import android.text.TextUtils;
 import android.util.Log;
 
 import com.android.internal.R;
@@ -45,11 +47,11 @@ public class PropImitationHooks {
     private static volatile boolean sIsGms = false;
     private static volatile boolean sIsFinsky = false;
 
-    public static void setProps(Application app) {
-        final String packageName = app.getPackageName();
-        final String processName = app.getProcessName();
+    public static void setProps(Context context) {
+        final String packageName = context.getPackageName();
+        final String processName = Application.getProcessName();
 
-        if (packageName == null || processName == null) {
+        if (TextUtils.isEmpty(packageName) || processName == null) {
             return;
         }
 
-- 
2.39.1

